Chat/assignment-chat/server-chat/src/main/java/ru/assignment/model/ChatModelListener.java

```diff
@@ -0,0 +1,12 @@
+package ru.assignment.model;
+
+import ru.assignment.message.ChatMessage;
+
+/**
+ * Created by Андрей on 19.02.2015.
+ */
+public interface ChatModelListener {
+    void sendMessageToClient(ChatMessage chatMessage);
+    void sendMessageToChatModel(String message);
```

Comment:
There shouldn't be any methods in listener which assume that listener will modify model state.
Listener only listens to model events
Session#listen() -> ChatModel -> listener#onNewMessage(Message message)